internal/testutils/provider.go

```diff
@@ -14,11 +15,23 @@ import (
 	"time"
 
 	"github.com/coreos/go-oidc/v3/oidc"
+	"github.com/golang-jwt/jwt/v5"
 	"github.com/ubuntu/authd-oidc-brokers/internal/consts"
 	"github.com/ubuntu/authd-oidc-brokers/internal/providers/info"
 	"golang.org/x/oauth2"
 )
 
+// MockKey is the RSA key used to sign the JWTs for the mock provider.
+var MockKey *rsa.PrivateKey
+
+func init() {
+	key, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		panic("Setup: Could not generate RSA key for the Mock")
```

Comment:
```go
panic(fmt.Sprintf("Setup: Could not generate RSA key for the Mock: %v"))
```

So in the unlikely case of failure (I see it happen more in some not-fully configured CI than in real world) we can get it better?

Reply:
+1 for

```suggestion
		panic(fmt.Sprintf("Setup: Could not generate RSA key for the Mock: %v", err))
```